test/test_linalg.py

```diff
@@ -5869,13 +5870,11 @@ def _test(m, k, n, transpose_a, transpose_b, test_equal=True):
 
     @unittest.skipIf(IS_WINDOWS, "Skipped on Windows!")
     @unittest.skipIf(IS_FBCODE and IS_REMOTE_GPU, "cublas runtime error")
+    @skipCUDAIfRocmVersionLessThan((6, 0))
     @onlyCUDA
     def test__int_mm_errors(self, device):
-        if TEST_WITH_ROCM:
-            self.skipTest("_int_mm not compiled for ROCM")
-
         version = _get_torch_cuda_version()
-        if version < (11, 7):
+        if version < (11, 7) and not TEST_WITH_ROCM:
             self.skipTest("_int_mm only compiled for CUDA 11.7")
```

Comment:
Perhaps the following instead?  No need to check `TEST_WITH_ROCM` since the torch cuda version doesn't make sense for rocm pytorch and you already have the skip decorator added to protect against rocm less than 6.0.  Checking for `torch.version.cuda` (or `torch.version.cuda is not None` is what we're really checking here, but the short version is ... shorter.)

```
if torch.version.cuda and _get_torch_cuda_version() < (11, 7):
    self.skipTest("_int_mm only compiled for CUDA 11.7")
```

Reply:
improve readability thanks